security/fma-interop-tx-handling.md

```diff
@@ -0,0 +1,137 @@
+which matches the specified index and content hash.
+
+Including an Executing Message which *does not* match to an Initiating Message is considered to be invalid to the protocol.
+The result is that the *entire block* which contains this Invalid Message is replaced, producing a reorg on the chain
+at this height.
+
+Because the consequenquence for an Invalid Message is so high, Chain Operators are highly incentivized to check Executing
+Messages before they are added to a block. *However*, being excessive with these checks can cause interruption to the
+chain's regular forward-progress. There must be a balance taken in checking messages.
+
+### Two Extremes
+
+To understand the purpose of these decisions, lets consider the extreme validity-checking policies we could adopt:
+
+**Check Every Message Exhaustively**
+- In this model, every Executing Message is checked constantly, at a maximum rate, and each message is checked as close to block-building as possible.
+- The compute cost to check every message during block building adds time to *every* transaction, blowing out our ability
+to build a block in under 2s.
+- The validity of the included Executing Messages would be *as correct* as possible. However! Even *after* the block is built, the data being relied upon (cross-unsafe data) could change on the Initiating Chain if they suffer a reorg.
+So while this policy is *most correct*, it is not *totally correct*
+
+**Never Check Any Messages**
+- In this model, we optimize only for not taking any additional compute tasks, instead just trusting every message.
+- Naturally, there is no impact into block building or any other process, BUT...
+- Blocks would easily become invalid, because an attacker could submit Invalid Messages, even just one per 2s, and prevent the Sequencer from ever building a valid block.
+
+So, no matter what solution we pick, we deal with *some* amount of uncertainty and take *some* amount of additional compute load.
+
+## The Solution Design
+
+The [Interop Topology and Tx Flow for Interop Chains Design Doc](https://github.com/ethereum-optimism/design-docs/pull/218)
+describes the solution design we plan on going with:
+
+- All Executing Message are checked once at `proxyd` ingress.
+- All Executing Message are checked once at Node Mempool ingress (not counting Sequencer).
+- All Executing Message in Node Mempools are Batched at checked on a regular interval.
+- If an Executing Message is ever Invalid, it is discarded and not retried.
+- *No* Checks are done at Block Building time.
+
+This FMA describes the potential negative consequences of this design. We have selected this design because it maximizes
+the opportunities for Invalid Messages to be caught and discarded, while also leaving the block building hot-path from
+having to take on new compute.
+
+# FMs:
+
+## FM1: Checks Fail to Catch an Invalid Executing Message
```

Comment:
Is this specific to ingress checks?

Reply:
I think we should have a section specific for ingress and a section specific for finalization. Both of these problems can have different solutions

Reply:
This FM is overlap with the Supervisor FM2a (not 2b). I am not sure if it needs to be kept in this document, but I wanted it around to demonstrate that if these layers fail, an invalid message gets into a block.

To your question about if it's specific to some check -- this would have to apply to *every* check layer, not just one. If ingress is failing but mempool checks are working, it still gets filtered. If mempool is failing but ingress is working, it gets filtered.

I'll rework this FM to be more descriptive of how all the layers could fail together.